cloverage/src/cloverage/instrument.clj

```diff
@@ -517,43 +546,48 @@
             instrumented-form (try
                                 (wrap f-var line-hint form)
                                 (catch Throwable e
-                                  (throw+ e "Couldn't wrap form %s at line %s"
-                                          form line-hint)))]
-        (eval-form filename form line-hint instrumented-form)
+                                  (throw (ex-info "Error instrumenting form"
+                                                  {:filename filename, :line line-hint, :form form}
+                                                  e))))]
+        (try
+          (eval-form filename form line-hint instrumented-form)
+          (catch Throwable e
+            (throw (ex-info "Error evaluating instrumented form" {} e))))
         instrumented-form)
-      ;; if we run into an error instrumenting a form, log it and return/eval the uninstrumented form, so we can
-      ;; continue
+      ;; if we run into an error instrumenting a form or evaluating it, log it and return/eval the uninstrumented
+      ;; form, so we can continue
       (catch Throwable e
-        (log/error "Error instrumenting form"
-                   (ex-info "Error instrumenting form" {:filename filename
-                                                        :line     line-hint
-                                                        :form     form}
-                            e))
+        (log/error (.getMessage e) "\n"
+                   (binding [*print-meta* true]
+                     (with-out-str (pprint/pprint (Throwable->map e)))))
         (eval-form filename form line-hint form)
         form))))
 
 (defn instrument-file
   "Instrument all the forms in a file. Returns sequence of instrumented forms."
   [f-var lib filename]
-  (with-open [^java.io.Closeable source-reader (s/form-reader lib)]
+  (with-open [source-reader (source/form-reader lib)]
     (transduce
      (map (partial instrument-form f-var filename))
      conj
-     []
-     (source-forms source-reader))))
+     (source/forms source-reader))))
 
 (defn instrument
-  "Instruments and evaluates a list of forms."
-  [f-var lib]
-  (let [filename (s/resource-path lib)]
+  "Instrument and evaluate the forms in the namespace named by `ns-symbol` using an instrumentation function referred to
+  by `f-var`. The instrumentation function has the signature a function with the signature
+
+    (f line-hint form) -> instrumented-form"
+
+  [f-var ns-symbol]
+  (let [filename (source/resource-path ns-symbol)]
     (try
-      (let [instrumented (instrument-file f-var lib filename)]
-        (d/dump-instrumented instrumented lib)
+      (let [instrumented (instrument-file f-var ns-symbol filename)]
+        (d/dump-instrumented instrumented ns-symbol)
```

Comment:
Personally I found the name `lib` confusing -- it actually refers to a symbol naming a namespace, but that's not obvious. Elsewhere the codebase uses `ns-symbol` so I changed the code here to use that for consistency.